src/custom/utils/index.ts

```diff
@@ -116,3 +116,43 @@ export function formatOrderId(orderId: string): string {
   // 0x is at index 0 of orderId, shorten. Else return id as is
   return has0x?.index === 0 ? shortenOrderId(orderId, 2, orderId.length) : orderId
 }
+
+export async function fetchData(endpoint: string, options?: RequestInit) {
+  let response
+  try {
+    response = await fetch(endpoint, options)
+    if (!response.ok) throw new Error('[fetchData]::Server returned non-ok response.')
+  } catch (error) {
+    console.error(error)
+    throw new Error(error)
+  }
+  return response
+}
+
+export async function promiseFirstResolved<T>(
+  promiseList: ((...args: any[]) => Promise<T | undefined>)[]
+  // fallback: T
+): Promise<{ id: number | null; result: T | null }> {
```

Comment:
You return the relative position in the `id` (index)
It's a bit obscure method signature

Reply:
The signature is hard to read. Maybe you can declare the return types or arguments, so its more declarative (u give them a name)

promiseList: is probably AsyncCall, or PromiseGenerator...